packages/react-vapor/src/components/modal/tests/ConfirmationModalProvider.spec.tsx

```diff
@@ -1,155 +1,134 @@
+import {renderModal} from 'react-vapor-test-utils';
 import {ConfirmationModalProvider} from '../ConfirmationModalProvider';
+import {ModalCompositeConnected} from '../ModalComposite';
 
 describe('<UnsavedChangeModalProvider/>', () => {
-    let confirmationModalProvider: ShallowWrapper;
-    let heavyConfirmationModalProvider;
-    let regularClickActionSpy: jest.Mock<any, any>;
-    let promptBeforeClickActionSpy: jest.Mock<any, any>;
-    const store = getStoreMock();
-    beforeEach(() => {
-        regularClickActionSpy = jest.fn();
-        promptBeforeClickActionSpy = jest.fn();
-    });
-
-    afterEach(() => {
-        regularClickActionSpy.mockReset();
-        promptBeforeClickActionSpy.mockReset();
-        if (confirmationModalProvider?.exists()) {
-            confirmationModalProvider.unmount();
-        }
-    });
-
-    const shallowMountUnsavedModalProvider = (shouldConfirm: boolean = false) =>
-        shallow(
-            <ConfirmationModalProvider
-                shouldConfirm={shouldConfirm}
-                modalTitle="Are you sure?"
-                modalBodyChildren={<div>some content</div>}
-            >
-                {({promptBefore}) => (
-                    <ModalCompositeConnected
-                        id="👾"
-                        title="🙄"
-                        classes={['mod-slide-in-bottom', 'mod-big', 'mod-stick-bottom']}
-                        modalBodyChildren="🙈 🙉 🙊"
-                        modalFooterChildren={
-                            <button
-                                className="btn"
-                                onClick={() => promptBefore(promptBeforeClickActionSpy) && regularClickActionSpy()}
-                            />
-                        }
-                        modalBodyClasses={['mod-header-padding', 'mod-form-top-bottom-padding']}
-                        docLink={{url: 'https://www.coveo.com', tooltip: {title: 'Go to coveo.com'}}}
-                    />
-                )}
-            </ConfirmationModalProvider>
-        );
-
-    const heavilyMountUnsavedModalProvider = (shouldConfirm: boolean = false) =>
-        mount(
-            <ConfirmationModalProvider
-                shouldConfirm={shouldConfirm}
-                modalTitle="Are you sure?"
-                modalBodyChildren={<div>some content</div>}
-            >
-                {({promptBefore}) => (
-                    <Provider store={store}>
+    describe('when shouldConfirm prop is true:', () => {
+        it('should open the ConfirmationModal to interrupt the leaving action if isDirty is true', () => {
+            renderModal(
+                <ConfirmationModalProvider
+                    shouldConfirm
+                    confirmationModalId="👾-unsaved-modal"
+                    modalTitle="Are you sure?"
+                    modalBodyChildren="Big bada boom"
+                >
+                    {({promptBefore}) => (
                         <ModalCompositeConnected
```

Comment:
I'm questioning the relevancy of rendering a modal in the test here. The confirmation modal provider doesn't care about what is rendered inside it so we could test its functionality with a simpler test.

```tsx
renderModal(
    <ConfirmationModalProvider
        shouldConfirm
        confirmationModalId="👾-unsaved-modal"
        modalTitle="Are you sure?"
        modalBodyChildren="Big bada boom"
    >
        {({promptBefore}) => (
            <button onClick={() => promptBefore(jest.fn())}>prompt confirmation modal</button>
        )}
    </ConfirmationModalProvider>
);
```

Reply:
I'm using the Modal since it has the `isDirty` prop

Reply:
It doesn't have the `isDirty` prop? At least its not defined in the ModalComposite props and is not used in its implementation. 😕